src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx

```diff
@@ -10,11 +13,53 @@ import SentryTypes from 'app/sentryTypes';
 
 import LoadingPanel from '../loadingPanel';
 
+type RenderProps = {
+  loading: boolean;
+  reloading: boolean;
+
+  // timeseries data
+  timeseriesData?: Series[];
+  allTimeseriesData?: EventsStatsData;
+  originalTimeseriesData?: EventsStatsData;
+  timeseriesTotals?: object;
+  originalPreviousTimeseriesData?: EventsStatsData | null;
```

Comment:
Can it really be be both undefined or null? Same for `previousTimeseriesData` and `timeAggregatedData` below

Reply:
Yeah, if data from API is not ready, these values are all undefined: https://github.com/getsentry/sentry/pull/14251/files/1a333953296fb3e3ca2f2169550a7afa2c1460e6#diff-a441d3d3ab295c283b5d790001e0c5a8R293 -- and it can be null if an option wasn't passed (e.g. include previous period).